From: Andrey Konovalov <andreyknvl@google.com>
Subject: lib/stackdepot: fix for "annotate racy pool_index accesses"
Date: Sat, 11 Feb 2023 16:48:42 +0100

Change the remaining reference to pool_index in stack_depot_fetch to
pool_index_cached.

Link: https://lkml.kernel.org/r/95cf53f0da2c112aa2cc54456cbcd6975c3ff343.1676129911.git.andreyknvl@google.com
Signed-off-by: Andrey Konovalov <andreyknvl@google.com>
Cc: Alexander Potapenko <glider@google.com>
Cc: Evgenii Stepanov <eugenis@google.com>
Cc: Marco Elver <elver@google.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/lib/stackdepot.c~lib-stackdepot-annotate-racy-pool_index-accesses-fix
+++ a/lib/stackdepot.c
@@ -521,7 +521,7 @@ unsigned int stack_depot_fetch(depot_sta
 
 	if (parts.pool_index > pool_index_cached) {
 		WARN(1, "pool index %d out of bounds (%d) for stack id %08x\n",
-			parts.pool_index, pool_index, handle);
+			parts.pool_index, pool_index_cached, handle);
 		return 0;
 	}
 	pool = stack_pools[parts.pool_index];
_
